### ISSUE TITLE
Fix S108: Rule should not report an issue in Razor cshtml

### ISSUE THREAD

**Issue by fosterwork** [SonarSource/sonar-csharp#687]
### Description

When a code block in a cshtml file only contains html, SonarAnalyzer incorrectly warns to "Either remove or fill this block of code"

### Repro steps

1. Step A
Create a .cshtml file

2. Step B
Add this code:
```
@if (1==2)
{
    <p>Test</p>
}
```

### Expected behavior

There should be no warning.

### Actual behavior

Warning is displayed: "Either remove or fill this block of code"

### Known workarounds

Disable SonarAnalyzer

### Related information

SonarAnalyzer.CSharp 6.3.0.2862
Visual Studio 2017

**Comment by Evangelink** [comment]
[RSPEC-108](https://jira.sonarsource.com/browse/RSPEC-108)

**Comment by Evangelink** [comment]
@fosterwork Could you provide a full cshtml (not too big if you can) reproducing the issue?
The following bit of code is not raising any issue:
```
<html>
<body>
@if (1==2)
{
    <p>Test</p>
}
</body>
</html>
```

**Comment by fosterwork** [comment]
That file triggers the error for me with Visual Studio Enterprise 2017 Version 15.2 (26430.12) and SonarLint for Visual Studio 3.4.0.1732
![s108](https://user-images.githubusercontent.com/10677553/29592098-1458c1be-8757-11e7-9c23-d22a6c24f19e.png)

**Comment by pauldbentley** [comment]
I get this error in a .NET Framework MVC Web Application

Steps to reproduce

1. create a new project
2. c# ASP.NET Web Application (.NET Framework)
3. .NET Framework 4.8
4. Empty
5. Add MVC 5 View Page (Razor)

```csharp
@if (true)
{
    <p>Test</p>
}
```

![Image](https://github.com/user-attachments/assets/460134e4-5860-4670-87e5-e071b66916f4)

**Comment by pavel-mikula-sonarsource** [comment]
Hi, @pauldbentley,

We've recently added support for Razor file analysis, so it is possible that some issues have re-appeared. As this issue is old and closed, can you please create a new thread here? 
https://community.sonarsource.com/c/rules-and-languages/fp/7

Feel free to mention link to this issue there.

Thank you

**Comment by Latisha19** [comment]
For inside my core MVC's .cshtml, it is not supported well, I tried to exclude them with the following link:

[Excluding .cshtml Files Doesn't Work - SonarQube for IDE / Visual Studio - Sonar Community](https://community.sonarsource.com/t/excluding-cshtml-files-doesnt-work/142394)

However, nothing works. (had updated to plugin version 8.30.0.15605)

What else should I do?

**Comment by pavel-mikula-sonarsource** [comment]
https://community.sonarsource.com/ is the best place to ask for help with SonarQube for IDE

**Comment by Latisha19** [comment]
@pavel-mikula-sonarsource Thanks for your guide!
